GitVersionCore.Tests/Helpers/GitTestExtensions.cs

```diff
@@ -39,6 +37,48 @@ public static Commit[] MakeCommits(this IRepository repository, int numCommitsTo
             .ToArray();
     }
 
+    public static Commit CreateFileAndCommit(this IRepository repository, string relativeFileName, DateTimeOffset dateTimeOffset = default(DateTimeOffset))
+    {
+        if (dateTimeOffset == default(DateTimeOffset))
+        {
+            dateTimeOffset = DateTimeOffset.Now;
+        }
+
+        var randomFile = Path.Combine(repository.Info.WorkingDirectory, relativeFileName);
+        if (File.Exists(randomFile))
+        {
+            File.Delete(randomFile);
+        }
+
+        File.WriteAllText(randomFile, Guid.NewGuid().ToString());
+
+        // GHK: 2015-01-18: I know it's very ugly, but somehow we need to retry here otherwise "there is nothing to commit"
+        var retryCount = 3;
```

Comment:
I already implemented random content, but we can also add random length as well. Doesn't libgit see a difference when the file contents are different but equal length?